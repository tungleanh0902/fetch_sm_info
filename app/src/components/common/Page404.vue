<template>
  <TmPage data-title="404" class="four-oh-four">
    <TmPart class="four-oh-four-top">
      <h2>Oops! We can't find the page you're looking for.</h2>
      <h3>Error code: 404</h3>
    </TmPart>
    <TmPart>
      <p>Here are some helpful links to get you out of here:</p>
      <ul>
        <li>
          <router-link
            :to="{ name: 'portfolio', params: { networkId: networkSlug } }"
          >
            Portfolio
          </router-link>
        </li>
        <li>
          <router-link
            :to="{ name: 'validators', params: { networkId: networkSlug } }"
          >
            Validators
          </router-link>
        </li>
        <li>
          <router-link
            :to="{ name: 'transactions', params: { networkId: networkSlug } }"
          >
            Transactions
          </router-link>
        </li>
        <li>
          <router-link
            :to="{ name: 'proposals', params: { networkId: networkSlug } }"
          >
            Proposals
          </router-link>
        </li>
      </ul>
    </TmPart>
  </TmPage>
</template>

<script>
import TmPage from "common/TmPage"
import TmPart from "common/TmPart"
import { mapGetters } from "vuex"
export default {
  name: `page-404`,
  components: {
    TmPage,
    TmPart,
  },
  computed: {
    ...mapGetters([`networkSlug`]),
  },
}
</script>

<style>
.four-oh-four {
  padding: 2rem;
  background: var(--black);
  border-left: 1px solid var(--bc-dim);
}

.four-oh-four h2 {
  color: var(--bright);
  font-size: var(--h1);
  font-weight: 500;
  line-height: 3rem;
}

.four-oh-four h3 {
  color: var(--text);
  font-size: var(--h3);
  font-weight: 500;
  margin-top: 0;
}

.four-oh-four ul {
  padding-top: 1rem;
}
</style>
