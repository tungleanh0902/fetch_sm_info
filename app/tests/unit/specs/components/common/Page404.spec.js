import { shallowMount } from "@vue/test-utils"
import Page404 from "src/components/common/Page404"

describe(`Page404`, () => {
  let wrapper, $store

  beforeEach(() => {
    $store = {
      getters: {
        networkSlug: "cosmos-hub",
      },
    }
    wrapper = shallowMount(Page404, {
      mocks: {
        $store,
      },
      stubs: [`router-link`],
    })
  })

  it(`should show the 404 page`, async () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`should show links to other pages`, () => {
    expect(wrapper.findAll(`router-link-stub`).length > 0).toBe(true)
  })
})
