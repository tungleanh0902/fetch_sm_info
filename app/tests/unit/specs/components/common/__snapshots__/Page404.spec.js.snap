// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 should show the 404 page 1`] = `
<tmpage-stub
  class="four-oh-four"
  data-title="404"
  emptysubtitle=""
  emptytitle="No data"
  loaderpath=""
>
  <tmpart-stub
    class="four-oh-four-top"
  >
    <h2>
      Oops! We can't find the page you're looking for.
    </h2>
     
    <h3>
      Error code: 404
    </h3>
  </tmpart-stub>
   
  <tmpart-stub>
    <p>
      Here are some helpful links to get you out of here:
    </p>
     
    <ul>
      <li>
        <router-link-stub
          to="[object Object]"
        >
          
          Portfolio
        
        </router-link-stub>
      </li>
       
      <li>
        <router-link-stub
          to="[object Object]"
        >
          
          Validators
        
        </router-link-stub>
      </li>
       
      <li>
        <router-link-stub
          to="[object Object]"
        >
          
          Transactions
        
        </router-link-stub>
      </li>
       
      <li>
        <router-link-stub
          to="[object Object]"
        >
          
          Proposals
        
        </router-link-stub>
      </li>
    </ul>
  </tmpart-stub>
</tmpage-stub>
`;
